.ci/integTestGen/src/integTestGen.jl

```diff
@@ -0,0 +1,265 @@
+    url::String
+    modified_url::String
+    env_var::String
+    PackageInfo(url, env_var) = new(url, "", env_var)
+end
+
+"""
+    extract_env_vars_from_git_message!(package_infos::AbstractDict{String, PackageInfo}, var_name = "CI_COMMIT_MESSAGE")
+
+    Parse the commit message, if set via variable (usual `CI_COMMIT_MESSAGE`) and set custom URLs.
+"""
+function extract_env_vars_from_git_message!(package_infos::AbstractDict{String, PackageInfo}, var_name = "CI_COMMIT_MESSAGE")
+    if haskey(ENV, var_name)
+        for line in split(ENV[var_name], "\n")
+            line = strip(line)
+            for pkg_info in values(package_infos)
+                if startswith(line, pkg_info.env_var * ": ")
+                    ENV[pkg_info.env_var] = SubString(line, length(pkg_info.env_var * ": ") + 1)
+                end
+            end
+        end
+    end
+end
+
+"""
+    modify_package_url!(package_infos::AbstractDict{String, PackageInfo})
+
+Iterate over all entries of package_info. If an environment variable exits with the same name like,
+the `env_var` entry, set the value of the environment variable to `custom_url`.
+"""
+function modify_package_url!(package_infos::AbstractDict{String, PackageInfo})
+    for package_info in values(package_infos)
+        if haskey(ENV, package_info.env_var)
+            package_info.modified_url = ENV[package_info.env_var]
+        end
+    end
+end
+
+"""
+    modified_package_name(package_infos::AbstractDict{String, PackageInfo})
+
+Read the name of the modified (project) package from the environment variable `CI_DEPENDENCY_NAME`.
+
+# Returns
+- The name of the modified (project) package
+"""
+function modified_package_name(package_infos::AbstractDict{String, PackageInfo})
+    for env_var in ["CI_DEPENDENCY_NAME", "CI_PROJECT_DIR"]
+        if !haskey(ENV, env_var)
+            error("Environment variable $env_var is not set.")
+        end
+    end
+
+    if !haskey(package_infos, ENV["CI_DEPENDENCY_NAME"])
+        package_name = ENV["CI_DEPENDENCY_NAME"]
+        error("Error unknown package name $package_name}")
+    else
+        return ENV["CI_DEPENDENCY_NAME"]
+    end
+end
+
+"""
+    depending_projects(package_name, package_prefix, project_tree)
+
+    Return a list of packages, which has the package `package_name` as a dependency. Ignore all packages, which does not start with `package_prefix`.
+
+# Arguments
+- `package_name::String`: Name of the dependency
+- `package_prefix::Union{AbstractString,Regex}`: If package name does not start with the prefix, do not check if had the dependency.
+- `project_tree=PkgDependency.builddict(Pkg.project().uuid, Pkg.project())`: Project tree, where to search the dependend packages. Needs to be a nested dict.
+                                                                             Each (sub-)project needs to be AbstractDict{String, AbstractDict}
+
+# Returns
+- `::AbstractVector{String}`: all packages, which have the search dependency
```

Comment:
```suggestion
- `::AbstractVector{String}`: all packages which have the search dependency
```

See https://www.grammarly.com/blog/comma/  "Commas with relative clauses"